Entity/Revision.php

```diff
@@ -344,8 +344,8 @@ public function __construct()
                 $ancestor = $a[0];
                 $this->deleted = $ancestor->deleted;
                 $this->draft = true;
-                unset($this->finalizedBy);
-                unset($this->finalizedDate);
+                $this->finalizedBy = null;
+                $this->finalizedDate = null;
```

Comment:
Violation of phpstan, as the finalizedBy cannot be null. As discussed we will set a new DateTime to Now() for these two fields. (Can you do this for this pull request).
I will take care of the tests for the "id" field on __clone of an object.

Reply:
See 4e7c6c92022a2e10de5568123d9872d036eb673b